cognite/seismic/protos/v1/seismic_service_datatypes.proto

```diff
@@ -211,3 +212,10 @@ message TraceExtent {
 message LineBasedExtent {
     repeated TraceExtent extents = 1;
 }
+
+message SurveyCoverageSource {
+    oneof source {
```

Comment:
https://developers.google.com/protocol-buffers/docs/proto3#enum - should maybe this be an enum?

Reply:
Yeah that's probably cleaner, will update

Reply:
Annoyingly enum just resolves to an integer, even with a message that wraps it. We need the optional state to allow for defaulting to whatever coverage source is available. We could put the optional state into the enum but that seems strange on the returning side since we always know the source

Reply:
Yeah, that's an unfortunate effect of a crucial difference between protobuf's enum semantics and Rust's: 
> The enum type isn't used directly as a field, because the Protobuf spec mandates that enumerations values are 'open', and decoding unrecognized enumeration values must be possible.
(From https://github.com/tokio-rs/prost#enumerations)
There are getters and setters to help a bit though.

Reply:
Anyway, if we have the choice between a tidy API for clients and a neat server-side implementation, the neat API wins in my opinion.